experiments/fgo_nn_splines.py

```diff
@@ -196,12 +205,43 @@ def fgo_nn_splines_train_loop(train_dataloader, val_dataloader=None, subseq_len
     results['n_epochs'] = n_epochs
     results['n_actual_epochs'] = 0
     results['subseq_len'] = subseq_len
+    trajectories_to_save = 3
+    results['num_val_traj'] = trajectories_to_save
+    window_size=100
 
     output_path = Path(output_path)
     output_path.mkdir(parents=True, exist_ok=True)
 
-    device = torch.device(device)
+    train_dataset_path = output_path / "train_dataset.pkl"
+    if train_dataset_path.is_file():
+        print(f"Find cached datasets here - {output_path}. Loading from cache.")
+        with open(train_dataset_path, 'rb') as f:
+            train_dataset = pickle.load(f)
+        with open(output_path / "val_dataset.pkl", 'rb') as f:
+            val_dataset = pickle.load(f)
+    else:
+        path_to_splines = output_path / "splines_for_experiment"
+        generate_batch_of_splines(path_to_splines, number_of_splines=50, n_control_points=10, n_pts_spline_segment=100, val_ratio=0.2, is_random=False)
+        train_dataset = Spline_2D_Dataset(path_to_splines, 
+                                    window=window_size,
+                                    sampling_rate=100,
+                                    subseq_len=subseq_len,
+                                    mode='regression',
+                                    enable_noise= not True)
+
+        
+
+        val_dataset = Spline_2D_Dataset(path_to_splines, window=window_size, subseq_len=subseq_len, enable_noise= not True, stage="val")
```

Comment:
sebseq_len here should be greater - ideally it should match the whole experiment length. in current version it is set to be the same is for training, by default it is equal to 2